src/services/monaco.service.ts

```diff
@@ -89,8 +89,18 @@ export class MonacoService {
   }
 
   static async loadMonaco() {
-    const monacoInstance = await loader.init();
-    window.Monaco = monacoInstance;
+    let monaco: typeof monacoAPI;
+
+    // JEST cannot bundle monaco-editor in test environment so we need to fetch that package from cdn
+    // in dev or production environment we will use bundled monaco-editor
+    if (process.env.NODE_ENV === 'test') {
+      monaco = await loader.init();
+    } else {
+      monaco = await import('monaco-editor');
+      loader.config({ monaco });
+    }
```

Comment:
I don't have much idea about the Monaco internals but every time I see code that needs to be changed to make tests work, it gives me shivers 😄 Isn't there really a way to make tests work without having to make this code aware of the tests?

Reply:
I tried to map the imports to the appropriate modules, but this didn't work at all. Jest is building the project before testing (I was surprised) making monaco-editor mapping difficult. You can look for issues on github or even on stackoverflow where a lot of people have problems with this package on the jest side. I know that such ifs should not be, but this is the simplest solution to the problem.